tests/arp/test_stress_arp.py

```diff
@@ -0,0 +1,168 @@
+                                    ip_snd=arp_request_ip,
+                                    ip_tgt=arp_request_ip,
+                                    hw_snd=arp_src_mac,
+                                    hw_tgt='ff:ff:ff:ff:ff:ff'
+                                )
+        testutils.send_packet(ptfadapter, intf1_index, pkt)
+    logger.info("Sending {} arp entries".format(ip_num))
+
+def genrate_ipv4_ip():
+    ipv4_addr = ip_network(unicode(ARP_BASE_IP), strict=False)
+    ptf_intf_ipv4_hosts = ipv4_addr.hosts()
+    return list(ptf_intf_ipv4_hosts)
+
+def test_ipv4_arp(duthost, garp_enabled, ip_and_intf_info, intfs_for_test, ptfadapter, get_function_conpleteness_level):
+    """
+    Send gratuitous ARP (GARP) packet sfrom the PTF to the DUT
+
+    The DUT should learn the (previously unseen) ARP info from the packet
+    """
+    normalized_level = get_function_conpleteness_level
+    if normalized_level is None:
+        normalized_level = "basic"
+
+    ipv4_avaliable = get_crm_resources(duthost, "ipv4_neighbor", "available") - get_crm_resources(duthost, "ipv4_neighbor", "used")
+    fdb_avaliable = get_crm_resources(duthost, "fdb_entry", "available") - get_crm_resources(duthost, "fdb_entry", "used")
+    pytest_assert(ipv4_avaliable > 0 and fdb_avaliable > 0, "Entries have been filled")
+
+    arp_avaliable = min(min(ipv4_avaliable, fdb_avaliable), ENTRIES_NUMBERS)
+
+    pytest_require(garp_enabled, 'Gratuitous ARP not enabled for this device')
+    ptf_intf_ipv4_hosts = genrate_ipv4_ip()
+    ptf_intf_ipv4_hosts = list(ptf_intf_ipv4_hosts)[1:arp_avaliable + 1]
+    _, _, intf1_index, _, = intfs_for_test
+
+    loop_times = LOOP_TIMES_LEVEL_MAP[normalized_level]
+
+    while loop_times > 0:
+        loop_times -= 1
+        add_arp(ptf_intf_ipv4_hosts, intf1_index, ptfadapter)
+
+        pytest_assert(wait_until(20, 1, 0, lambda: get_fdb_dynamic_mac_count(duthost) >= arp_avaliable),
+                      "ARP Table Add failed")
+
+        clear_dut_arp_cache(duthost)
+        fdb_cleanup(duthost)
+
+        time.sleep(5)
+
+def generate_link_local_addr(mac):
+    parts = mac.split(":")
+    parts.insert(3, "ff")
+    parts.insert(4, "fe")
+    parts[0] = "{:x}".format(int(parts[0], 16) ^ 2)
+
+    ipv6Parts = []
+    for i in range(0, len(parts), 2):
+        ipv6Parts.append("".join(parts[i:i+2]))
+    ipv6 = "fc02:1000::{}".format(":".join(ipv6Parts))
+    return ipv6
+
+
+def packets_for_test(duthost, config_facts, tbinfo, ip_and_intf_info, fake_src_mac, fake_src_addr):
+    _, _, src_addr_v6, _, _ = ip_and_intf_info
+    fake_src_mac = fake_src_mac
+    vlans = config_facts['VLAN']
+    topology = tbinfo['topo']['name']
+    dut_mac = ''
+    for vlan_details in vlans.values():
+        if 'dualtor' in topology:
+            dut_mac = vlan_details['mac'].lower()
+        else:
+            dut_mac = duthost.shell('sonic-cfggen -d -v \'DEVICE_METADATA.localhost.mac\'')["stdout_lines"][0].decode(
```

Comment:
You can directly use `dut_mac = duthost.facts['router_mac']` here.

Reply:
I find this variable will not be used anymore, so I delete it.